app/kubernetes-ingress-controller/2.2.x/guides/using-gateway-api.md

```diff
@@ -0,0 +1,196 @@
+
+[Gateway API](https://gateway-api.sigs.k8s.io/) is a new set of resources for
+configuring networking in Kubernetes. It expands on Ingress to configure
+additional types of routes (TCP, UDP, and TLS in addition to HTTP/HTTPS),
+support backends other than Service, and manage the proxies that implement
+routes.
+
+Gateway API and Kong's implementation of Gateway API are both alpha stage and
+under active development. Features and implementation specifics will change
+before their initial general availability release.
+
+## Enabling the feature
+
+The Gateway API CRDs are not yet available by default in Kubernetes. You must
+first [install them](https://gateway-api.sigs.k8s.io/v1alpha2/guides/getting-started/#installing-gateway-api-crds-manually).
+
+The default controller configuration disables Gateway API handling. To enable
+it, set `ingressController.env.feature_gates: Gateway=true` in your Helm
+values.yaml (set `CONTROLLER_FEATURE_GATES=Gateway=true` if not using Helm).
+Note that you must restart Pods with this flag set _after_ installing the
+Gateway API CRDs.
+
+If using Helm, you must use chart version 2.7 or higher. Older versions do not
+include the ServiceAccount permissions necessary for KIC to read Gateway API
+resources.
+
+## Testing Connectivity to Kong
+
+This guide assumes that the `PROXY_IP` environment variable is
+set to contain the IP address or URL pointing to Kong.
+Please follow one of the
+[deployment guides](/kubernetes-ingress-controller/{{page.kong_version}}/deployment/overview) to configure this environment variable.
+
+If everything is setup correctly, making a request to Kong should return
+HTTP 404 Not Found.
+
+{:.note}
+> **Note**: If you are running the example using Minikube on MacOS, you may need 
+to run [`minikube tunnel`](https://minikube.sigs.k8s.io/docs/handbook/accessing/#loadbalancer-access)
+in a separate terminal window.  This exposes LoadBalancer services 
+externally, which is not enabled by default.
+
+```bash
+$ curl -i $PROXY_IP
+HTTP/1.1 404 Not Found
+Content-Type: application/json; charset=utf-8
+Connection: keep-alive
+Content-Length: 48
+Server: kong/1.2.1
+
+{"message":"no Route matched with those values"}
+```
+
+This is expected as Kong does not yet know how to proxy the request.
+
+## Set up an echo-server
+
+Setup an echo-server application to demonstrate how
+to use the {{site.kic_product_name}}:
+
+```bash
+$ kubectl apply -f https://bit.ly/echo-service
+```
+
+## Add a GatewayClass and Gateway
+
+The Gateway resource represents the proxy instance that handles traffic for a
+set of Gateway API routes, and a GatewayClass describes characteristics shared
+by all Gateways of a given type.
+
+
+```bash
+$ echo "apiVersion: gateway.networking.k8s.io/v1alpha2
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: GatewayClass
+metadata:
+  name: kong
+spec:
+  controllerName: konghq.com/kic-gateway-controller
+" | kubectl apply -f -
+
+gatewayclass.gateway.networking.k8s.io/kong created
```

Comment:
Please either comment out the response or put it in a separate codeblock. We keep getting feedback/complaints that users can't just copy/paste the command as-is due to the extra line of text.

Usually something like this works:

> Response:
> ```
> gatewayclass.gateway.networking.k8s.io/kong created
> ```

Applies to all examples in this guide.